doc/src/atomvm-tooling.md

```diff
@@ -0,0 +1,471 @@
+
+For example:
+
+        ###########################################################
+
+           ###    ########  #######  ##     ## ##     ## ##     ##
+          ## ##      ##    ##     ## ###   ### ##     ## ###   ###
+         ##   ##     ##    ##     ## #### #### ##     ## #### ####
+        ##     ##    ##    ##     ## ## ### ## ##     ## ## ### ##
+        #########    ##    ##     ## ##     ##  ##   ##  ##     ##
+        ##     ##    ##    ##     ## ##     ##   ## ##   ##     ##
+        ##     ##    ##     #######  ##     ##    ###    ##     ##
+
+        ###########################################################
+
+    I (852) AtomVM: Starting AtomVM revision 0.6.0-alpha.1
+    I (862) sys: Loaded BEAM partition boot.avm at address 0x1d0000 (size=262144 bytes)
+    I (882) network_driver: Initialized network interface
+    I (882) network_driver: Created default event loop
+    I (902) AtomVM: Found startup beam esp32init.beam
+    I (922) AtomVM: Starting esp32init.beam...
+    ---
+    AtomVM init.
+    I (932) sys: Loaded BEAM partition main.avm at address 0x210000 (size=1048576 bytes)
+    Starting application...
+    Hello World
+    AtomVM finished with return value: ok
+    I (972) AtomVM: AtomVM application terminated.  Going to sleep forever ...
+
+#### STM32
+
+To flash AtomVM AVM file to an STM32 device, use the `stm32_flash` target in the `atomvm` namespace.
+
+> Note. In order to use the `stm32_flash` target, you will need to install the  `st-flash` tool from the open source (bsd-3 licensed) [stlink](https://github.com/stlink-org/stlink) suite of stm32 utilities.
+
+Since the AtomVM core libraries are not flashed to an STM32 device, you will need to include is library in your application.  As part of the build process for the STM32, you will have built the AtomVM core libraries into a file named `atomvmlib.avm`
+
+> Note.  It is critical that the version of the AtomVM core libraries match the version of the AtomVM virtual machine you built as part of the STM32 build.  Be sure to use the version of this library (written to `build/lib/atomvmlib.avm` during the build process).  For more information about how to build AtomVM for the STM32 platform, see the AtomVM [Build Instructions](build-instructions.md).
+
+In general, it is also a good idea to use the prune option when creating your application's AVM file.  This way, only the modules that are needed for your application will be included, which will decrease the size of your application's AVM file, leading to faster development times.
+
+Edit the `rebar.config` so that it includes the following `atomvm_rebar3_plugin` stanza, if it does not already.
+
+    {atomvm_rebar3_plugin, [
+        {packbeam, [prune]}
+    ]}.
+
+This stanza will guarantee that the generated packbeam file will be pruned when created.
+
+You may now flash your application to your STM32 device:
+
+    shell$ rebar3 atomvm stm32_flash -e /path/to/atomvmlib.avm
```

Comment:
Excellent! I did not realize that options passed down to dependency providers. This is very helpful.

Reply:
Indeed!  Run `DEBUG=1 rebar3 atomvm …` and you will see the effective config (based on command line, rebar.config, env var, and defaults, in that precendence order) in the debug log!